### PR TITLE
Switch to dompurify for sanitizing markdown content

### PR DESCRIPTION
**Pull request by mjbvz**
Switches us from using `insane` to instead use `dompurify`, which seems to be better maintained and also has some nice features, such as built-in trusted types support

I've tried to port over our existing sanitizer settings as best as possible, but there's not always a 1:1 mapping between how insane works and how dompurify does. I'd like to get this change in early in the iteration to catch potential regressions

### CODE REVIEW

**Comment by mjbvz** [comment]
Assigning:

@jrieken For trusted types
@alexr00 For sanitization of markdown strings (specifically the allowed styles)
@JacksonKearl For welcome page

**Review by alexr00 — APPROVED** [review]

File: src/vs/workbench/contrib/markdown/common/markdownDocumentRenderer.ts [hunk 1]
```diff
@@ -204,7 +198,10 @@ export async function renderMarkdownDocument(
 		marked(text, { highlight }, (err, value) => err ? reject(err) : resolve(value));
 	}).then(raw => {
 		if (shouldRemoveEmbeddedSVGs) {
-			return removeEmbeddedSVGs(raw);
+			const x = sanitize(raw);
+			console.log(raw);
+			console.log(x);
```
Comment (jrieken) [inline src/vs/workbench/contrib/markdown/common/markdownDocumentRenderer.ts:203]: `console.log`

**Review by jrieken — APPROVED** [review]
lgtm

**Comment by mjbvz** [comment]
@jrieken Any idea on why I now get the eslint error in `dom.ts`:

```
 [ban-element-innerhtml-assignments] Assigning directly to Element#innerHTML can result in XSS vulnerabilities.
```

The code looks like:

```
		const html = dompurify.sanitize(value, { ...options, RETURN_TRUSTED_TYPE: true });
		node.innerHTML = html as any;
```

Where `html` is a `TrustedHtml` object

We were previously assigning to `innerHTML` as well, so I'm not sure why this error seems to have just started showing up. Also not sure how to fix it

**Comment by jrieken** [comment]
> We were previously assigning to innerHTML as well, so I'm not sure why this error seems to have just started showing up. Also not sure how to fix it

@mjbvz https://github.com/microsoft/vscode/pull/131950/commits/fd122aad63ba0654262c4606d2bb173a92f93fc5 isn't the best way to tackle this. A special sequence of casts is needed here... `XYZ as unknown as string`, see https://github.com/microsoft/vscode/issues/106396#issuecomment-692625393 for more details

**Comment by jrieken** [comment]
fyi - I have pushed two commits

* e837c9b which is not great but better than the exemptions 
* a6cc04876e0a9685e6fd22aac0350eea723729cc makes the monaco-esm build happy (hopefully 🤞 )

**Comment by mjbvz** [comment]
Thanks everyone! Merging in to get testing in insiders to catch potential regressions

**Pull request merged**